### ISSUE TITLE
Ability to resolve lowest compatible versions of specific packages only

### ISSUE THREAD

**Issue by connortann** [astral-sh/uv#5492]
First, thank you for the amazing package! We're migrating to it over at the `shap` repo and seeing wonderful speedups.

My question is about how best to test for compatibility with the lowest supported "important" dependencies, whilst using the highest versions of all "dev" dependencies. I have tried a few ways so far, but each has a few issues.

## A typical scenario

Taking [shap](https://github.com/shap/shap) as an example, we have about 10 dependencies for which we'd like to test against old versions. These are the major scientific python libraries for which we'd like to ensure we support a reasonably large range: numpy, pandas, tensorflow, pytorch, et cetera. It's quite likely that users will have specific versions of these packages in their environments, so we want to avoid overly restrictive pins.

However, we also have a larger number of other "dev" dependencies (like pytest, setuptools, ipython) for which we only really need to use the latest version.

In our case, both of these groups are mostly _optional_ dependencies for our package, in the "test" extra. We install them on ci with `uv pip install --system .[test]`

## Options explored so far

I tried the `--resolution=lowest-direct` flag to test older versions of numpy and tensorflow. However, it also resolves the lowest versions of all the "dev" dependencies like pytest. This leaves us with a few options, none of which are ideal:
1. Add lower pins for every single dev dependency. These would then have to be maintained manually; AFAIK there is no automated solution like dependabot to update the lower pins according to something like SPEC 0.
2. Add exact pins for every dev dependency. Again, this is not ideal as they will have to be manually maintained and bumped for every release. 

I also noticed the `--exclude-newer` flag from #2088, which could be used to automatically determine the lowest versions that should be supported according to SPEC 0. However, this also restricts the versions of dev and build libraries. In our case, our build fails as we use recent features from `setuptools-scm` that weren't around two years ago.

## Proposal

It would be fantastic to be able to install the old versions of **just the subset of libraries** that we think are likely be to be present in user's environments, excluding the dev dependencies. Something like:

```bash
uv pip install --resolution=lowest-subset "numpy,pandas,tensorflow` .
```

Do you have any guidance on what might be a sensible approach?

**Comment by charliermarsh** [comment]
This seems like a reasonable use-case (though not yet sure how best to resolve it). One other option is that you could add a constraints file to set lower-bounds on your development dependencies... You could even take the output of the "standard" resolution to seed those constraints.

**Comment by antonymilne** [comment]
Just to add another data point and explain my use case: I'm looking for exactly the same thing on Vizro. We use hatch for project management, and I have an environment `lower-bounds`. This includes both dev dependencies and some of our core dependencies with manually specified lower bounds:

```
[envs.default]
dependencies = ["pytest"] # and others

[envs.lower-bounds]
# inherits dev dependencies from default environment and then manually specify lower bounds
# for some core dependencies
extra-dependencies = ["pydantic==1.10.13", "dash==2.17.1"]
```

What I'd like to do is this:
```
[envs.lower-bounds]
env-vars = {UV_RESOLUTION = "lowest-direct"}
```
... but unfortunately this means specifying a lower bound for all our dev dependencies. So ideally there would be some way to specify `lowest-direct` for just a subset of the dependencies.

Also just wanted to echo a huge thank you for the amazing work on uv. It's sped up our development process massively!
